file/types.go

```diff
@@ -533,14 +536,14 @@ type FServicePackage struct {
 	Versions    []FServiceVersion `json:"versions,omitempty" yaml:"versions,omitempty"`
 }
 
-// id is used for sorting.
-func (s FServicePackage) id() string {
-	if s.ID != nil {
-		return *s.ID
-	}
+// sortKey is used for sorting.
+func (s FServicePackage) sortKey() string {
 	if s.Name != nil {
 		return *s.Name
 	}
+	if s.ID != nil {
+		return *s.ID
+	}
 	return ""
 }
```

Comment:
The sort key for FServiceVersion seems to be missing and we should add that.
It can be done separately or part of this PR - either is fine.

Reply:
There was neither any  `compareID()` associated with this type, it seems like it was never sorted

Reply:
I've added https://github.com/Kong/deck/issues/348 to track it.
If you are interested, please take that on once you this PR is merged in.
Thanks you!